code/game/machinery/bots/secbot.dm

```diff
@@ -876,3 +876,192 @@ Auto Patrol: []"},
+							"Nobody breaks the law on my watch!")
+						speak(chase_message)
+					var/turf/olddist = get_dist(src, src.target)
+					walk_to(src, src.target,1,4)
+					if((get_dist(src, src.target)) >= (olddist))
+						src.frustration++
+					else
+						src.frustration = 0
+			else
+				src.frustration = 8
+
+		if(SECBOT_PREP_ARREST)		// preparing to arrest target
+
+			// see if he got away
+			if(get_dist(src, src.target) > 1)
+				src.anchored = 0
+				mode = SECBOT_HUNT
+				return
+			var/arrest_message = pick(
+				"Remember, crime doesn't pay!",
+				"Use your words, not your fists!",
+				"When in doubt, talk it out.",
+				"The weed of crime bears bitter fruit.",
+				"Just say \"No!\" to space drugs!",
+				"Violence is never the answer.",
+				"I'm not an officer, I'm a Security <em>monitor</em>.")
+			src.speak(arrest_message)
+
+			mode = SECBOT_IDLE
+			src.target = null
+			src.anchored = 0
+			src.last_found = world.time
+			src.frustration = 0
+
+		if(SECBOT_ARREST)		// arresting
+
+			src.anchored = 0
+			mode = SECBOT_IDLE
+			return
+
+
+		if(SECBOT_START_PATROL)	// start a patrol
+			if(path != null)
+				if(path.len > 0 && patrol_target)	// have a valid path, so just resume
+					mode = SECBOT_PATROL
+					return
+
+				else if(patrol_target)		// has patrol target already
+					spawn(0)
+						calc_path()		// so just find a route to it
+						if(path.len == 0)
+							patrol_target = 0
+							return
+						mode = SECBOT_PATROL
+
+
+				else					// no patrol target, so need a new one
+					find_patrol_target()
+					speak("Engaging patrol mode.")
+
+
+		if(SECBOT_PATROL)		// patrol mode
+			patrol_step()
+			spawn(5)
+				if(mode == SECBOT_PATROL)
+					patrol_step()
+
+		if(SECBOT_SUMMON)		// summoned to PDA
+			patrol_step()
+			spawn(4)
+				if(mode == SECBOT_SUMMON)
+					patrol_step()
+					sleep(4)
+					patrol_step()
+
+	return
+
+//Cheapsky Construction
+
+/obj/item/weapon/secbot_assembly/cheapsky
+	name = "box/signaler assembly"
+	desc = "Some sort of bizarre assembly."
+	icon = 'icons/obj/aibots.dmi'
+	icon_state = "box_signaler"
+	item_state = "syringe_kit"
+	build_step = 0
+	created_name = "Officer Cheapsky"
+
+/obj/item/clothing/head/cardborg/attackby(var/obj/item/device/assembly/signaler/S, mob/user)
+	..()
+	if(!issignaler(S))
+		return
+
+	if(S.secured)
+		qdel(S)
+		var/obj/item/weapon/secbot_assembly/cheapsky/A = new /obj/item/weapon/secbot_assembly/cheapsky
+		user.put_in_hands(A)
+		to_chat(user, "You add the signaler to \the [src].")
+		user.drop_from_inventory(src)
+		qdel(src)
+
+/obj/item/weapon/secbot_assembly/cheapsky/attackby(obj/item/weapon/W, mob/user)
+	if(W.sharpness && W.sharpness_flags & SHARP_BLADE && (!src.build_step))
+		src.build_step++
+		src.overlays += image('icons/obj/aibots.dmi', "bs_hole")
+		to_chat(user, "You cut a hole in \the [src]!")
+
+	else if(isprox(W) && (src.build_step == 1))
+		if(user.drop_item(W))
+			to_chat(user, "You complete the Securitron! Beep boop.")
```

Comment:
Officer Cheapsky is *a* securitron. Though I suppose I could change it to `"You complete \the [S]!"`